openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -62,6 +62,74 @@ def _get_resname(off_mol) -> str:
     return names[0]
 
 
+def _validate_alchemical_components(
+        alchemical_components: dict[str, list[Component]],
+        mapping: Optional[dict[str, ComponentMapping]],
+):
+    """
+    Checks that the alchemical components are suitable for the RFE protocol.
+
+    Specifically we check:
+      1. That all alchemical components are mapped.
+      2. That all alchemical components are SmallMoleculeComponents.
+      3. That the mappings don't involve element changes in core atoms
+         (note: this is a temporary limitation).
+
+    Parameters
+    ----------
+    alchemical_components : dict[str, list[Component]]
+      Dictionary contatining the alchemical components for
+      states A and B.
+    mapping : dict[str, ComponentMapping]
+      Dictionary of mappings between transforming components.
+
+    Raises
+    ------
+    ValueError
+      * If there are more than one mapping or mapping is None
+      * If there are any unmapped alchemical components.
+      * If there are any alchemical components that are not
+        SmallMoleculeComponents.
+      * Mappings which involve element changes in core atoms.
+    """
+    # Check mapping
+    # For now we only allow for a single mapping, this will likely change
+    if mapping is None or len(mapping.values()) > 1:
+        errmsg = "A single LigandAtomMapping is expected for this Protocol"
+        raise ValueError(errmsg)
+
+    # Check that all alchemical components are mapped & small molecules
+    mapped = {}
+    mapped['stateA'] = [m.componentA for m in mapping.values()]
+    mapped['stateB'] = [m.componentB for m in mapping.values()]
+
+    for idx in ['stateA', 'stateB']:
+        if len(alchemical_components[idx]) != len(mapped[idx]):
+            errmsg = f"missing alchemical components in {idx}"
+            raise ValueError(errmsg)
+        for comp in alchemical_components[idx]:
+            if comp not in mapped[idx]:
+                raise ValueError(f"Unmapped alchemical component {comp}")
+            if not isinstance(comp, SmallMoleculeComponent):  # pragma: no-cover
```

Comment:
in general for non-public stuff I'd just carve out functionality as we need it rather than try and be clever, maybe I'd have left a todo that this is where an extra code branch might have to be